lib/src/over_react_test/prop_type_utils.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:html';
+import 'dart:js';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/component_base.dart';
+import 'package:over_react_test/jacket.dart';
+import 'package:react/react_client/react_interop.dart' show PropTypes, ReactElement;
+import 'package:test/test.dart';
+
+/// Tests the prop types of a component by calling the props, adding children if necessary, and checking the console
+/// for expected log output.
+///
+/// By default the test will validate that the console error count matches an
+/// inferred expected count (inferred via the length of
+/// customErrorMessageList and whether or not the `render()` will throw).
+/// This is useful for simple tests, but may be turned off for complex
+/// componentry tests where the console may log unrelated messages.
+///
+/// If a Component's `render()` will throw because it is relying on a prop
+/// that was not passed in correctly, that can be caught by setting
+/// `willThrow` to true and passing in the expected Error Matcher.
+///
+/// If there are multiple `propType` checks and you wish to test a
+/// combination, pass in multiple expected errors into
+/// `customErrorMessageList`. The order of the messages in the List should
+/// match the order the component will catch them in (e.i. the order of the
+/// checks declared in the component).
+void testPropTypesWithUiProps(
```

Comment:
I think this is a great start, but I'm a little worried about this API being restrictive in the kinds of errors you can test for.

Inspired by the [`prints`](https://pub.dev/documentation/test_api/latest/test_api/prints.html) matcher and the build_test package's [`recordLogs`](https://pub.dev/documentation/build_test/latest/build_test/recordLogs.html), here are some ideas:

What if we did something (similar to `prints`) that gives you more flexibility around what you want to do:
```dart
expect(() => mount(Foo()()), logsPropTypeWarning(contains('foo is required')));
```
```dart
expect(() => menu.show(), logsReactWarning(contains('missing key')));
```
```dart
expect(() => jacket.rerender(), logsPropTypeWarnings([
  contains('foo is required'),
  contains('bar must be specified when baz is specified'),
]));
```
```dart
expect(() => mount(Foo()()), logsNoPropTypeWarnings);
```

And then expose the underlying API (similar to `recordLogs`) that allows async or other advanced use-cases:
```dart
final logs = recordLogs(() {
  click();
  something();
});
expect(logs.firstWhere(contains('foo')), contains('because bar'));
```
```dart
final logsStream = recordLogsAsync(() async {
  click();
  await something();
});
expect(logsStream, emitsReactWarning(contains('foo')));
```